boa/src/syntax/ast/punc.rs

```diff
@@ -169,6 +169,12 @@ impl Punctuator {
     }
 }
 
+impl Into<BinOp> for Punctuator {
+    fn into(self) -> BinOp {
+        self.as_binop().expect("Could not get binary operation")
```

Comment:
Once again, this should not fail. We can implement `TryInto` if we still need something like this.

Reply:
Ah yeah, `TryInto` is definitely more correct here.

The reasoning for impl'ing these traits is to abstract the main portion of the `parse` method exposed by the `expression!` macro (in which `as_binop` is called) into a generic helper that is generic over the `op` type - since before it was only a Punctuator, it now needs to have Keyword support.

So, both Punctuator and Keyword are impl'ing `Into<BinOp>` (or now TryInto<BinOp>) in order to support both Keyword and Punctuator in the main `match` over the token.

This method and usage can be found [here](https://github.com/jasonwilliams/boa/pull/350/files#diff-805f77066eac33b82808bcee94de20d7R73) - I think that my explanation will make more sense while also seeing the implementation 😄